Setup.bat

```diff
@@ -27,14 +36,70 @@ call :MarkStartOfBlock "Setup the git hooks"
 call :MarkEndOfBlock "Setup the git hooks"
 
 call :MarkStartOfBlock "Check dependencies"
-    if not defined UNREAL_HOME (
-        echo Error: Please set UNREAL_HOME environment variable to point to the Unreal Engine folder.
+
+    rem Find the Unreal Engine used to build this project.
+    set UNREAL_ENGINE=""
+
+    rem Get the Unreal Engine used by this project by querying the registry for the engine association found in the .uproject.
+    for /f "delims=" %%A in (' powershell -Command "Get-Childitem -Path %ProjectDirectory% -Recurse -Include *.uproject -File | %% {$_.FullName}" ') do set UPROJECT="%%A"

```

Comment:
Isn't it true that the .uproject won't always be in the project directory? (ie. it'll normally live in a `Game` etc. folder inside the project directoy)

Reply:
Correct, the `-Recurse`argument here will search all the folders within the `ProjectDirectory` to find the folder where the `.uproject` exists (such as `Game` in our example)